src/write.rs

```diff
@@ -204,8 +229,10 @@ impl<W: Write + io::Seek> ZipWriter<W> {
             files: Vec::new(),
             stats: Default::default(),
             writing_to_file: false,
-            comment: String::new(),
+            writing_to_extra_field: false,
+            writing_to_central_extra_field_only: false,
```

Comment:
You mean to add more variants to `GenericZipWriter` instead of using the `writing_` flags? This is tricky for `writing_to_extra_field` as it can be `true` for all variants except `Closed` and would need to be a parameter of each.